### PR TITLE
[Spark] Simplify ray worker node killing logic when parent process died

### PR DESCRIPTION
**Pull request by WeichenXu123**
## Why are these changes needed?

Simplify ray worker node killing logic when parent process died.

Previously, we registered `PR_SET_PDEATHSIG` event via subprocess `preexec_fn` argument, this way is complex and it relies on "libc.so.6" library that is only available on Linux system.

In current PR, it detects parent died event via `def check_parent_alive()` funtion, it checks whether the parent process ID changed, if changed, if means original parent process is died and the process becomes **orphan process** (orphan process has parent process of pid=1). This way can support any UNIX system such as macOS.

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/util/spark/start_ray_node.py [hunk 1]
```diff
@@ -109,7 +110,31 @@ def try_clean_temp_dir_at_exit():
             fcntl.flock(lock_fd, fcntl.LOCK_UN)
             os.close(lock_fd)
 
+
```
Comment (jjyao) [inline python/ray/util/spark/start_ray_node.py:113]: Update the above comments.

**Review by jjyao — APPROVED** [review]

File: python/ray/util/spark/start_ray_node.py [hunk 2]
```diff
@@ -109,11 +109,37 @@ def try_clean_temp_dir_at_exit():
             fcntl.flock(lock_fd, fcntl.LOCK_UN)
             os.close(lock_fd)
 
+    def check_parent_alive() -> None:
+        orig_parent_id = os.getppid()
```
Comment (jjyao) [inline python/ray/util/spark/start_ray_node.py:113]: Actually there can be a race condition that when we execute this line, the parent process already died?

Should we just pass the ppid to the `start_ray_node` child process?

Comment (WeichenXu123) [reply python/ray/util/spark/start_ray_node.py:113]: This could be safer. Good suggestion.

File: python/ray/util/spark/cluster_init.py [hunk 3]
```diff
@@ -41,6 +40,7 @@
 MAX_NUM_WORKER_NODES = -1
 
 RAY_ON_SPARK_COLLECT_LOG_TO_PATH = "RAY_ON_SPARK_COLLECT_LOG_TO_PATH"
+START_RAY_PARENT_PID = "START_RAY_PARENT_PID"
```
Comment (jjyao) [inline python/ray/util/spark/cluster_init.py:43]: RAY_ON_SPARK_START_RAY_PARENT_PID to minimizing the chance of naming conflict.

**Pull request merged**